static/scripts/onboarding/onboarding.ts

```diff
@@ -27,102 +28,19 @@ const REPO_NAME = "ubiquibot-config";
 const DEFAULT_REPO = "ubiquibot";
 const KEY_PATH = ".github/ubiquibot-config.yml";
 const DEFAULT_PATH = "ubiquibot-config-default.json";
-const KEY_NAME = "private-key-encrypted";
 const KEY_PREFIX = "HSK_";
 const X25519_KEY = "5ghIlfGjz_ChcYlBDOG7dzmgAgBPuTahpvTMBipSH00";
 
 let encryptedValue = "";
 
-interface ConfLabel {
-  name: string;
-}
-
-interface CommandLabel {
-  name: string;
-  enabled: boolean;
-}
-
-interface IIncentive {
-  comment: {
-    elements: Record<string, unknown>;
-    totals: {
-      word: number;
-    };
-  };
-}
-
-interface IControl {
-  label: boolean;
-  organization: boolean;
-}
-
-interface IConf {
-  "private-key-encrypted"?: string;
-  "safe-address"?: string;
-  "base-multiplier"?: number;
-  "auto-pay-mode"?: boolean;
-  "analytics-mode"?: boolean;
-  "max-concurrent-bounties"?: number;
-  "incentive-mode"?: boolean;
-  "evm-network-id"?: number;
-  "price-multiplier"?: number;
-  "issue-creator-multiplier"?: number;
-  "payment-permit-max-price"?: number;
-  "max-concurrent-assigns"?: number;
-  "assistive-pricing"?: boolean;
-  "disable-analytics"?: boolean;
-  "comment-incentives"?: boolean;
-  "register-wallet-with-verification"?: boolean;
-  "promotion-comment"?: string;
-  "default-labels"?: string[];
-  "time-labels"?: ConfLabel[];
-  "priority-labels"?: ConfLabel[];
-  "command-settings"?: CommandLabel[];
-  incentives?: IIncentive;
-  "enable-access-control"?: IControl;
-}
-
-let defaultConf: IConf = {
-  "private-key-encrypted": "",
-  "safe-address": "",
-  "base-multiplier": 1,
-  "auto-pay-mode": false,
-  "analytics-mode": false,
-  "max-concurrent-bounties": 1,
-  "incentive-mode": false,
-  "evm-network-id": 1,
-  "price-multiplier": 1,
-  "issue-creator-multiplier": 1,
-  "payment-permit-max-price": 1,
-  "max-concurrent-assigns": 1,
-  "assistive-pricing": false,
-  "disable-analytics": false,
-  "comment-incentives": false,
-  "register-wallet-with-verification": false,
-  "promotion-comment": "",
-  "default-labels": [],
-  "time-labels": [],
-  "priority-labels": [],
-  "command-settings": [],
-  incentives: {
-    comment: {
-      elements: {},
-      totals: {
-        word: 0,
-      },
-    },
-  },
-  "enable-access-control": {
-    label: false,
-    organization: true,
-  },
-};
+let defaultConf: BotConfig;
 
-export const parseYAML = async (data: any): Promise<any | undefined> => {
+export const parseYAML = async <T>(data: string | undefined) => {
```

Comment:
Since you're importing the submodule wouldn't it make sense to import this function from the bot repository so that the parsing logic always stays in sync?

Reply:
It would absolutely make sense, but the file with the Yaml logic inside `generate-configuration.ts` imports a lot of libraries and files not in the project. Either we include these in the package.json of pay.ubq.fi project, or we have to move it out in another file from the project ubiquibot and make sure it doesn't import any third party